### ISSUE TITLE
Enable DNS-over-HTTPS by default, instead of in auto-upgrade mode

### ISSUE THREAD

**Issue by pag-crypto** [brave/brave-browser#22369]
## Description 

DNS-over-HTTPS is only available in auto-upgrade mode in Brave; this is behavior inherited from Chromium. DNS-over-HTTPS should instead be the default choice for DNS resolution, as it is in Firefox. This will improve the privacy of many Brave users' DNS queries.

## Steps to Reproduce

   1. Open Brave browser in network where DoH cannot be auto-upgraded.
   2. Send DNS query
   

## Actual result:

DNS query is sent in the clear.

## Expected result:

DNS query sent encrypted with DoH.

## Reproduces how often: 

Easily reproduced

## Brave version (brave://version info)

all

## Miscellaneous Information:

Yan Zhu asked me to file this issue. I've never filed an issue with Brave before, so it's possible I've done something wrong - happy to change this if so.

**Comment by diracdeltas** [comment]
thanks paul (this is yan) - we'll discuss this!

**Comment by ice155** [comment]
_"This will improve the privacy of many Brave users' DNS queries."_: While this can be right, it can also be wrong. Not everyone is okay with their DNS queries being sent to a DNS resolver that they did not choose. If DNS over HTTPS (DoH) will from now on be enabled by default, I am guessing that you will do the same Mozilla did with Firefox: use Cloudflare's DoH resolver. But what if the user is not aware of that and they do not want to use Cloudflare's DoH resolver? Then their DNS queries will be sent to a resolver they did not choose. I think it's best to leave it disabled by default, the user will enable it if they want to. An idea: Maybe you could ask the user a question on a new/fresh install, whether they want to enable it or not, and that it secures their DNS queries.

**Comment by NumDeP** [comment]
Hi @ice155 I was thinking the same thing. Do you know if it is enabled in Chrome by default? If it is, then I would assume it is likely going to be pushed via Google's own infrastructure and if isn't enabled yet and Google decides to push it later, this is something Brave would naturally want to de-google and what should the team do, take it off temporarily on the next update just to give users a choice, it doesn't seem ideal to me.

I'm curious, assuming Brave and Firefox DNS is On DoH by default, if you have two computer systems that are exactly the same and on one of them you use Firefox to head to lets say netflix.com and on the other computer you use Brave, what literally differentiates the queries? My thoughts are if both browsers are practically utilizing the same third-party infrastructure i.e Cloudflare being the likely bet because its technology is arguable one the best, not forgetting I think they were one of the first few to start the conversation and development of DoH, surely it's better to have hundreds of millions of queries randomly pooled if there isn't anything particularly identifiable if you know what I mean.

I think it's just more efficient encrypted and I think most people wouldn't want to fiddle about with such things. Brave being a privacy based browser, it stands to reasons that it be enabled by default. I take the point of giving a choice upon a fresh install but I feel like Brave just does that way too much already. Sorry for being laborious.

**Comment by TylerDurden23** [comment]
I don't think it is a good idea to do it by default. My expectation would be, that if I have configured a DNS server in my network, this DNS server will be used - and not that my DNS requests are routed to some centralized entity like cloudflare or google.

**Comment by buawf** [comment]
If selected Cloudlfare 1.1.1.1 via the drop-down options in secure dns, what is the DoH url brave is making ?
